augur/refine.py

```diff
@@ -2,15 +2,15 @@
 Refine an initial tree using sequence metadata.
 """
 import numpy as np
-import os, shutil, time, sys
+import sys
 from Bio import Phylo
 from .utils import read_metadata, read_tree, get_numerical_dates, write_json, InvalidTreeError
-from treetime.vcf_utils import read_vcf, write_vcf
+from treetime.vcf_utils import read_vcf
 from treetime.seq_utils import profile_maps
 
 def refine(tree=None, aln=None, ref=None, dates=None, branch_length_inference='auto',
              confidence=False, resolve_polytomies=True, max_iter=2, precision='auto',
-             infer_gtr=True, Tc=0.01, reroot=None, use_marginal=False, fixed_pi=None,
+             infer_gtr=True, Tc=0.01, reroot=None, use_marginal='always', fixed_pi=None, use_fft=True,
```

Comment:
Does it make sense that default in CLI is `joint` but refine here has function default `use_marginal=always`?

Which default overrides the other one? Presumably CLI if entry is through CLI, but not if `refine` is called for directly?

What happens if `use_marginal=False` and `use_fft=True`? Will `refine` then still use fft or not?

Reply:
I wasn't too worried about the difference in defaults here, since the CLI always passes through its own arguments for `use_marginal` and `use_fft`, but I do think there is potential confusion about when FFT is actually getting used or not. I expected that kind of logic (or any warnings/error about incompatible arguments) to live in TreeTime's internals, so Augur isn't any more tightly coupled to TreeTime...